### ISSUE TITLE
Can't add clients on BOA3

### ISSUE THREAD

**Issue by bserem** [omega8cc/boa#926]
When trying to add a new client (and thus a system user) using BOA3, Aegir throws an error (The website encountered an unexpected error. Please try again later. ) and does nothing.

This is true both for new installations and updates from v2.4.9

**Comment by omega8cc** [comment]
It is a known problem with Aegir 3, please leave the email fields empty when adding new client: https://www.drupal.org/node/2670716

It will create the user with SSH/SFTP access, but without backend (Aegir) access, until the underlying problem will be fixed.

**Comment by Pls** [comment]
Yep, I just encountered this issue and found temp fix. This is related with userprotect module. Disabling "Auto-protect new-users" option from /admin/config/people/userprotect/protection_defaults seems to solve this issue. Of course, userprotect module has to be fixed to solve this issue properly.

**Comment by omega8cc** [comment]
@Pls -- Interesting! Thanks for sharing this.

**Comment by omega8cc** [comment]
While disabling "Auto-protect new-users" option in BOA allows to add the client with email address, it only exposes the problem in Aegir, not in the user protect module.

Just try to add another user with the same machine name.

You will see "The website encountered an unexpected error. Please try again later." again, and the error will be "The machine-readable name is already in use. It must be unique.", instead of "Client internal name already in use, try..." etc.

**Comment by bserem** [comment]
Thanks for all the info

**Comment by gandhiano** [comment]
I get this error on the dblog, if it is of any help:

```
Message	PDOException: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'up_delete' in 'field list': INSERT INTO {userprotect} (uid, up_type, up_roles, up_name, up_mail, up_pass, up_status, up_openid, up_delete, up_edit) VALUES (:db_insert_placeholder_0, :db_insert_placeholder_1, :db_insert_placeholder_2, :db_insert_placeholder_3, :db_insert_placeholder_4, :db_insert_placeholder_5, :db_insert_placeholder_6, :db_insert_placeholder_7, :db_insert_placeholder_8, :db_insert_placeholder_9); Array ( [:db_insert_placeholder_0] => 9 [:db_insert_placeholder_1] => user [:db_insert_placeholder_2] => 1 [:db_insert_placeholder_3] => 0 [:db_insert_placeholder_4] => 0 [:db_insert_placeholder_5] => 0 [:db_insert_placeholder_6] => 0 [:db_insert_placeholder_7] => 0 [:db_insert_placeholder_8] => 0 [:db_insert_placeholder_9] => 0 ) in userprotect_add_user() (line 789 of /data/disk/o1/aegir/distro/009/profiles/hostmaster/modules/contrib/userprotect/userprotect.module).
```

Disabling the userprotect effectively overcomes the issue. What does exactly this do, and what does it mean to have it disabled until a fix comes?

**Comment by macmladen** [comment]
I'd add that problem actually lies in that module should let

`The root user specifically bypasses all protections.`

but that does **not** seem to work. So you can go to `/admin/config/people/userprotect/protection_defaults` and allow Administrator role **bypass** and you will be able to add client (maybe selecting **e-mail** only could be enough).

**Comment by omega8cc** [comment]
The problem can't be fixed by disabling this protection or granting bypass privileges. Again, the real problem is with Hostmaster (Aegir) which uses Drupal 7 core procedures here, instead of its own. There could be further problems with the module itself, but it is better to have the feature temporarily "broken" (partially), than to allow to ignore the protection, since otherwise in a multi-user instances, with more than one "admin", it could allow to either escalate privileges or break things.

**Comment by gandhiano** [comment]
This issue is still existing and makes it really difficult to add new users/clients. I have tried to add a client without e-mail address, and this works. But I also need to add an user and associate it to the client, as to provide access to the Aegir control panel. This seems not to be working any longer, even the normal user creation.

**Comment by omega8cc** [comment]
Yes, we known that it's still not fixed, as the issue status confirms.

**Comment by bserem** [comment]
Strangely enough, adding a new client (without an email address) doesn't generate a system user for me anymore.

**Comment by omega8cc** [comment]
SSH account is generated once you make the client an owner of at least one site.

**Comment by omega8cc** [comment]
That said, we aim to fix this problem in 3.2.0

**Comment by omega8cc** [comment]
The real, hidden in the Octopus custom feature problem was trivial: https://github.com/omega8cc/hosting_custom_settings/commit/e10e1ce7ef6d7e2da7ac4b8daf38677ac45cb93a

Thanks @gandhiano for the exact hint which helped to find the culprit!

Thanks all for the useful feedback and patience!

We have also fixed related problem: https://www.drupal.org/node/2850435
